core/src/layers/prometheus.rs

```diff
@@ -88,12 +88,62 @@ use crate::*;
 #[derive(Default, Debug, Clone)]
 pub struct PrometheusLayer {
     registry: Registry,
+    requests_duration_seconds_buckets: Vec<f64>,
+    bytes_total_buckets: Vec<f64>,
 }
 
 impl PrometheusLayer {
     /// create PrometheusLayer by incoming registry.
+    /// For promtheus layer, we will use default buckets for HistogramMetric.
+    /// Be careful, if you want to use custom buckets, please use [`PrometheusLayer::with_registry_and_custom_metric_bucket`].
+    /// Or set buckets directly by [`PrometheusLayer::requests_duration_seconds_buckets`] and [`PrometheusLayer::bytes_total_buckets`] if you need it.
     pub fn with_registry(registry: Registry) -> Self {
-        Self { registry }
+        let mut result = Self {
+            registry,
+            ..Default::default()
+        };
+        result.requests_duration_seconds_buckets = exponential_buckets(0.01, 2.0, 16).unwrap();
+        result.bytes_total_buckets = exponential_buckets(0.01, 2.0, 16).unwrap();
+        result
+    }
+
+    /// create PrometheusLayer by incoming registry and custom buckets.
+    pub fn with_registry_and_custom_metric_bucket(
+        registry: Registry,
+        requests_duration_seconds_buckets: Vec<f64>,
+        bytes_total_buckets: Vec<f64>,
+    ) -> Self {
+        let mut result = Self {
+            registry,
+            ..Default::default()
+        };
+        result.requests_duration_seconds_buckets = if requests_duration_seconds_buckets.is_empty() {
+            exponential_buckets(0.01, 2.0, 16).unwrap()
+        } else {
+            requests_duration_seconds_buckets
+        };
+        result.bytes_total_buckets = if bytes_total_buckets.is_empty() {
+            exponential_buckets(0.01, 2.0, 16).unwrap()
+        } else {
+            bytes_total_buckets
+        };
+        result
+    }
+
+    /// set buckets for requests_duration_seconds
+    pub fn requests_duration_seconds_buckets(&mut self, buckets: Vec<f64>) -> &mut Self {
```

Comment:
It's by design that to take `mut self` and returns `Self`. The API style makes it's possible to write code like:

```rust
let layer = PrometheusLayer::with_registry(reg)
  .requests_duration_seconds_buckets(buckets);

xxx(layer);
```

instead of:

```rust
let mut layer = PrometheusLayer::with_registry(reg)
layer.requests_duration_seconds_buckets(buckets);

xxx(layer);
```

Reply:
But here may some chain call, like layer.requests_duration_seconds_buckets(buckets).bytes_total_buckets(buckets)

I think mut Self more semantic?

Reply:
> I think mut Self more semantic?

`mut self` is semantic too. Other layers also follow the same style.

Reply:
SGTM